### PR TITLE
Improve game mainLoop safety

### PR DESCRIPTION
**Pull request by pandamicro**

### CODE REVIEW

**Comment by minggo** [comment]
测试了，还是一样的效果。

File: cocos/core/game.ts [hunk 1]
```diff
@@ -709,29 +696,34 @@ export class Game extends EventTarget {
     }
     // Run game.
     private _runMainLoop () {
-        if (EDITOR && !legacyCC.GAME_VIEW) {
+        if (!this._inited || (EDITOR && !legacyCC.GAME_VIEW)) {
```
Comment (pandamicro) [inline cocos/core/game.ts:695]: @minggo  能不能看一下从后台恢复的时候 this._inited 是 false 还是 true，如果还在插屏阶段，应该是 false

Comment (minggo) [reply cocos/core/game.ts:695]: 是 true

**Review by hana-alice — APPROVED** [review]
tested. seems good to me, need someone else to review.

**Review by JayceLai — APPROVED** [review]

**Review by SantyWang — APPROVED** [review]

**Pull request merged**